compiler/x/codegen/X86BinaryEncoding.cpp

```diff
@@ -176,21 +176,36 @@ uint8_t getMemoryBarrierBinaryLengthLowerBound(int32_t barrier, TR::CodeGenerato
 
 // -----------------------------------------------------------------------------
 // OMR::X86::Instruction:: member functions
+bool OMR::X86::Instruction::needsLockPrefix()
+   {
+   return self()->getOpCode().needsLockPrefix();
+   }
 
-uint8_t *OMR::X86::Instruction::generateBinaryEncoding()
+uint8_t* OMR::X86::Instruction::generateBinaryEncoding()
    {
-   // *this    swipeable for debugging purposes
-   uint8_t *instructionStart = self()->cg()->getBinaryBufferCursor();
-   uint8_t *cursor           = instructionStart;
-   if (self()->getOpCode().needsRepPrefix())
+   for(;;)
```

Comment:
The need for this infinite loop is not immediately clear to me and I think dserves a comment since I would have thought an instruction would encode once and be done?